ArcGISRuntimeSDKQt_CppSamples/Scenes/OpenMobileScenePackage/OpenMobileScenePackage.cpp

```diff
@@ -0,0 +1,150 @@
+// [WriteFile Name=OpenMobileScenePackage, Category=Scenes]
+// [Legal]
+// Copyright 2018 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "OpenMobileScenePackage.h"
+
+#include "MobileScenePackage.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+
+#include <QDir>
+#include <QDateTime>
+#include <QtCore/qglobal.h>
+
+#ifdef Q_OS_IOS
+#include <QStandardPaths>
+#endif // Q_OS_IOS
+
+using namespace Esri::ArcGISRuntime;
+
+OpenMobileScenePackage::OpenMobileScenePackage(QObject* parent /* = nullptr */):
+  QObject(parent)
+{  
+}
+
+OpenMobileScenePackage::~OpenMobileScenePackage() = default;
+
+// helper method to get cross platform data path
+namespace {
+  QString defaultHomePath()
+  {
+    QString homePath;
+
+  #ifdef Q_OS_ANDROID
```

Comment:
This counts for this and the point cloud sample - but you can break down all these methods for a writable location into two (provided without testing):

```
QString defaultHomePath()
{
#if defined(Q_OS_ANDROID) || defined(Q_OS_IOS)
  return QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation);
#else 
  return QStandardPaths::writableLocation(QStandardPaths::HomeLocation);
#endif
}
```

Reply:
I would also suggest not calling `defaultHomePath` as this is not a home path on devices.

Reply:
I'm up for the rename, but I think the path refactoring might be a bigger change than I want to do right now. We have a consistent story about where data gets downloaded, and that is baked into the json and is used by the viewer. This seems like it would minimally change the story for Android. This new approach at least improves on what we did before by excluding the Extras dependency for C++. Seems like there is further room for refinement

Reply:
I think these are the same directory on Android (`/sdcard` and the user directory `/storage/emulated/0/`).